pkg/kubelet/container_gc.go

```diff
@@ -157,6 +170,18 @@ func (cgc *realContainerGC) GarbageCollect() error {
 		}
 	}
 
+	// Remove dead symlinks - should only happen on upgrade
+	// from a k8s version without proper log symlink cleanup
+	logSymlinks, _ := filepath.Glob(path.Join(cgc.containerLogsDir, "*.log"))
+	for _, logSymlink := range logSymlinks {
+		if _, err = os.Stat(logSymlink); os.IsNotExist(err) {
```

Comment:
As far as I can determine, this won't work, as the stat() call on the dead symlink will succeed?

e.g. (and yes, the results are the same under linux):

```
4 lrwxr-xr-x 1 quinton eng   8 Jul  7 13:43 test.sym -> test.txt
0 -rw-r--r-- 1 quinton eng   0 Jul  7 13:42 test.txt
quinton-macbookpro:symlink-test quinton$ stat test.sym
  File: ‘test.sym’ -> ‘test.txt’
  Size: 8         	Blocks: 8          IO Block: 4096   symbolic link
Device: 1000004h/16777220d	Inode: 64971777    Links: 1
Access: (0755/lrwxr-xr-x)  Uid: (157215/ quinton)   Gid: ( 5000/     eng)
Access: 2015-07-07 13:43:05.000000000 -0700
Modify: 2015-07-07 13:43:05.000000000 -0700
Change: 2015-07-07 13:43:05.000000000 -0700
 Birth: 2015-07-07 13:43:05.000000000 -0700
quinton-macbookpro:symlink-test quinton$ rm test.txt
quinton-macbookpro:symlink-test quinton$ stat test.sym
  File: ‘test.sym’ -> ‘test.txt’
  Size: 8         	Blocks: 8          IO Block: 4096   symbolic link
Device: 1000004h/16777220d	Inode: 64971777    Links: 1
Access: (0755/lrwxr-xr-x)  Uid: (157215/ quinton)   Gid: ( 5000/     eng)
Access: 2015-07-07 13:43:05.000000000 -0700
Modify: 2015-07-07 13:43:05.000000000 -0700
Change: 2015-07-07 13:43:05.000000000 -0700
 Birth: 2015-07-07 13:43:05.000000000 -0700
```

Am I missing something?